WebDriverAgentLib/Routing/FBSession.m

```diff
@@ -181,6 +181,9 @@ - (XCUIApplication *)activeApplication
       if ([FBConfiguration shouldRespectSystemAlerts]
           && [XCUIApplication.fb_systemApplication descendantsMatchingType:XCUIElementTypeAlert].count > 0) {
         return XCUIApplication.fb_systemApplication;
+      } else if ([FBConfiguration shouldRespectHalfModals]
```

Comment:
I would say shouldRespectSystemAlerts and shouldRespectHalfModals settings should not be exclusive. If both options are set to true and both conditions must be checked

Reply:
Thank you for your suggestion! Does https://github.com/appium/WebDriverAgent/pull/946/commits/00c4fa8ce2b9f10e712220ac74dafe5553eeb418 address your suggestion?